app/controllers/ansible_variables_controller.rb

```diff
@@ -49,11 +50,19 @@ def resource_params
   end
 
   def new_vars
-    params[:changed] ? params[:changed][:new].as_json : {}
+    fetch_vars :new
   end
 
   def old_vars
-    params[:changed] ? params[:changed][:obsolete].as_json : {}
+    fetch_vars :obsolete
+  end
+
+  def updated_vars
+    fetch_vars :update
+  end
+
+  def fetch_vars(key)
+    params[:changed] ? (params[:changed][key].as_json || {}) : {}
```

Comment:
since you touch this already, `params.fetch(:changed, {}).fetch(key).try(:as_json) || {}` is I think equivalent and slightly safer (in case `params[:changed]` is not a hash)